src/include/gurt/debug_setup.h

```diff
@@ -0,0 +1,206 @@
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define D_LOG_DECLARE_DB(flag, s_name, l_name, mask, arg)	\
+	extern d_dbug_t	flag;
+
+/** Instantiate a debug flag variable
+ *
+ * \param flag		The name of the variable used to store the value of the
+ *			debug bit.
+ * \param s_name	Unused
+ * \param l_name	Unused
+ * \param mask		Unused
+ * \param arg		Unused
+ */
+#define D_LOG_INSTANTIATE_DB(flag, s_name, l_name, mask, arg)	\
+	d_dbug_t	flag;
+
+/** Internal use macro as part of D_LOG_REGISTER_DB.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define _D_LOG_ALLOCATE_DBG_BIT(dbgbit, s_name, l_name, mask, rc)	\
+	rc = d_log_dbg_bit_alloc(&dbgbit, #s_name, #l_name);		\
+	if (rc < 0)	{						\
+		rc = -DER_UNINIT;					\
+		D_PRINT_ERR("Could not get debug bit " #s_name "\n");	\
+		break;							\
+	}
+
+/** Internal use macro as part of D_LOG_DEREGISTER_DB.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define _D_LOG_DEALLOCATE_DBG_BIT(dbgbit, s_name, l_name, mask, rc)	\
+	rc = d_log_dbg_bit_dealloc(#s_name);				\
+	if (rc < 0)	{						\
+		rc = -DER_UNINIT;					\
+		D_PRINT_ERR("Could not free debug bit " #s_name "\n");	\
+		break;							\
+	}
+
+#define D_LOG_REGISTER_DB(db_foreach)					\
+	({								\
+		int __rc = 0;						\
+		do {							\
+			db_foreach(_D_LOG_ALLOCATE_DBG_BIT, __rc)	\
+		} while (0);						\
+		__rc;							\
+	})
+
+#define D_LOG_DEREGISTER_DB(db_foreach)					\
+	({								\
+		int __rc = 0;						\
+		do {							\
+			db_foreach(_D_LOG_DEALLOCATE_DBG_BIT, __rc)	\
+		} while (0);						\
+		__rc;							\
+	})
+
+/** These macros are used intended to be used with FOREACH macros that define
+ *  log facilities in your library.   It will utilize internal FOREACH macros
+ *  to define debug bits as well as user defined macros.  See
+ *  D_FOREACH_GURT_FAC for an example.
+ *
+ * These macros can also be used standalone but are not quite as convenient
+ * due to required but unused arguments
+ */
+
+/** Declare an extern to global facility variable
+ *
+ * \param s_name	Short name for the facility
+ * \param l_name	Unused
+ * \param user_dbg_bits	Unused
+ */
+#define D_LOG_DECLARE_FAC(s_name, l_name, user_dbg_bits)	\
+	extern int _D_LOG_FAC_DECL(s_name);			\
+
+/** Declare a global facility variable
+ *
+ * \param s_name	Short name for the facility
+ * \param l_name	Unused
+ * \param user_dbg_bits	Unused
+ */
+#define D_LOG_INSTANTIATE_FAC(s_name, l_name, user_dbg_bits)	\
+	int _D_LOG_FAC_DECL(s_name);				\
+
+
+/** Internal use macro as part of D_LOG_REGISTER_FAC.  It's multiline on purpose
+ *  so we can break from the foreach loop.  Checkpatch will complain but don't
+ *  see a better way.
+ */
+#define _D_LOG_ALLOCATE_LOG_FACILITY(s_name, l_name, rc)		\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop